### PR TITLE
Feature/staked chart for weekly

### PR DESCRIPTION
**Pull request by dmitrylineup**
- added tabs for weekly profit dialog
 - added staked area chart for Weekly profit
 - empty values generated for missed time in each data series
________________________________________________________________
 - added cleansing parts for options data
 - added platforms distributor

### CODE REVIEW

File: src/app/modules/dialogs/weekly-profit-dialog/weekly-profit-dialog.component.ts [hunk 1]
```diff
@@ -32,6 +108,76 @@ export class WeeklyProfitDialogComponent extends ChartGeneralMethodsComponent im
...
+
+      let seriesCounter = 0;
+      this.option.legend.data = [];
+      this.option.series = [];
+      for (const key in groups) {
+          if (groups.hasOwnProperty(key)) {
+              this.option.series.push({
+                  name: key,
+                  type: 'line',
+                  stack: '总量',
```
Comment (belbix) [inline src/app/modules/dialogs/weekly-profit-dialog/weekly-profit-dialog.component.ts:158]: ?

File: src/app/modules/dialogs/weekly-profit-dialog/weekly-profit-dialog.component.ts [hunk 2]
```diff
@@ -32,6 +108,76 @@ export class WeeklyProfitDialogComponent extends ChartGeneralMethodsComponent im
...
+      this.loadSecondChart(data);
+
     });
   }
+
+  getPlatformFromVaultName(vaultName: string): string {
+      if (vaultName.startsWith('V_')) {
+          vaultName = vaultName.replace('V_', '');
+      }
+      const underscoreCount = vaultName.split('_').length;
```
Comment (belbix) [inline src/app/modules/dialogs/weekly-profit-dialog/weekly-profit-dialog.component.ts:120]: I had mistake when provided this method
use
      const underscoreCount = vaultName.split('_')?.length ?? 0;
      let platform = 'SINGLE';
      if (underscoreCount > 1) {

File: src/app/modules/dialogs/weekly-profit-dialog/weekly-profit-dialog.component.ts [hunk 3]
```diff
@@ -32,6 +108,76 @@ export class WeeklyProfitDialogComponent extends ChartGeneralMethodsComponent im
...
+                  data: []
+              });
+              const newDataList = [];
+
+              for (const i of times) {
+                  let value = 0;
+                  for (const j of groups[key]) {
+                      const currentDate = this.getCurrentDate(j);
+                      if (i === currentDate) {
+                          value = j.weeklyAllProfit / 1000;
```
Comment (belbix) [inline src/app/modules/dialogs/weekly-profit-dialog/weekly-profit-dialog.component.ts:172]: let's use fullRewardUsdTotal cumulative values
fullRewardUsdTotal - it is a cumulative sum of progits for 1 vault in USD

**Review by dmitry2199 — CHANGES_REQUESTED** [review]
In general looks good, but some changes are required to be done: 
* It is definitely needs to update tabs rendering, because current tabs implementation is very simple and non extendable.
* Also I'm wondering why you've changed `.loading-screen` styles? If it is used only for dev purpose, then please remove it.

**Review by dmitry2199 — APPROVED** [review]

**Review by dmitry2199 — CHANGES_REQUESTED** [review]
Changed to ON HOLD according to conversation with @belbix

**Pull request closed without merging**